upload-s3-artifact/action.sh

```diff
@@ -40,13 +41,14 @@ main() {
     echo "No file or directory at path '$INPUT_TARGET_PATH'"
     exit 1
   fi
+  aws_s3_key_prefix="${INPUT_AWS_S3_KEY_PREFIX:-""}"
   checksum="$(sha256sum "$file_path" | cut -d' ' -f1)"
   filename="$(basename "$file_path")"
   # NOTE: Get the last extension, if any
   extension="$(echo "$filename" | sed -n 's/^.*\.\(.*\)$/\1/p')"
   default_aws_s3_key="$checksum${extension:+.}${extension:-}"
   aws_s3_key="${INPUT_AWS_S3_KEY:-$default_aws_s3_key}"
-  aws s3 cp "$file_path" "s3://$INPUT_AWS_S3_BUCKET_NAME/$aws_s3_key"
+  aws s3 cp "$file_path" "s3://$INPUT_AWS_S3_BUCKET_NAME/$aws_s3_key_prefix$aws_s3_key"
```

Comment:
I think we need to wrap the variables in curlies here, and that this may be why the file was uploaded with the wrong name during [the test](https://github.com/capralifecycle/actions-lib/actions/runs/7959117903/job/21725438005?pr=7), although I'd expect to see a `_prefix` in the middle, hm..,

```sh
Completed 158 Bytes/158 Bytes (13.9 KiB/s) with 1 file(s) remaining
upload: ../../../../../tmp/act-upload-s3-artifact-directory/target.zip to s3://actions-lib-artifacts/directory.zipdirectory.zip
```

```suggestion
  aws s3 cp "$file_path" "s3://$INPUT_AWS_S3_BUCKET_NAME/${aws_s3_key_prefix}${aws_s3_key}"
```

Reply:
That was because of a bug in action.yaml:

`        INPUT_AWS_S3_KEY: ${{ inputs.aws-s3-key }}
        INPUT_AWS_S3_KEY_PREFIX: ${{ inputs.aws-s3-key-**prefix** }}` (prefix was not included)
`  

But I'm going to wrap the variables in curlies too for readability